Std/Data/Rat/Lemmas.lean

```diff
@@ -298,3 +298,17 @@ theorem inv_def (a : Rat) : a.inv = a.den /. a.num := by
   simp [inv_def, divInt_mul_right zg]
 
 theorem div_def (a b : Rat) : a / b = a * b.inv := rfl
+
+theorem ofScientific_true_def : Rat.ofScientific m true e = mkRat m (10 ^ e) := by
+  unfold Rat.ofScientific
+  exact normalize_eq_mkRat (Nat.ne_of_gt <| Nat.pos_pow_of_pos _ (by decide))
+
+theorem ofScientific_false_def : Rat.ofScientific m false e = (m * 10 ^ e : Nat) := by
+  unfold Rat.ofScientific
+  rfl
```

Comment:
```suggestion
  unfold Rat.ofScientific; apply normalize_eq_mkRat

theorem ofScientific_false_def : Rat.ofScientific m false e = (m * 10 ^ e : Nat) := by
  unfold Rat.ofScientific; rfl
```

Reply:
It didn't like `apply normalize_eq_mkRat`; the alternative along that direction is `unfold Rat.ofScientific; apply normalize_eq_mkRat ∘ Nat.ne_of_gt ∘ Nat.pos_pow_of_pos _; decide`, but I thought that was less clear than an `exact`.

Reply:
no you shouldn't be providing the proof at all here, it is supposed to be found by unification on the existing term. did you try `rw [normalize_eq_mkRat]`?

Reply:
Ah gotcha. Yup, that followed by a `rfl` worked.